Rlabkey/R/labkey.pipeline.R

```diff
@@ -0,0 +1,135 @@
+
+    url <- paste(baseUrl, "pipeline-analysis", folderPath, "getSavedProtocols.api", sep="")
+    response <- labkey.post(url, toJSON(params, auto_unbox=TRUE))
+
+    return (fromJSON(response, simplifyVector=FALSE, simplifyDataFrame=FALSE))
+}
+
+labkey.pipeline.getFileStatus <- function(baseUrl=NULL, folderPath, taskId, protocolName, path, files)
+{
+    baseUrl=labkey.getBaseUrl(baseUrl)
+
+    ## check required parameters
+    if (missing(baseUrl) || is.null(baseUrl)) stop (paste("A value must be specified for baseUrl."))
+    if (missing(folderPath)) stop (paste("A value must be specified for folderPath."))
+    if (missing(taskId) || is.null(taskId)) stop (paste("A value must be specified for taskId."))
+    if (missing(protocolName) || is.null(protocolName)) stop (paste("A value must be specified for protocolName."))
+    if (missing(path) || is.null(path)) stop (paste("A value must be specified for path."))
+    if (missing(files) || is.null(files)) stop (paste("A value must be specified for files."))
+
+    ## check parameter types
+    if (!is.list(files)) stop (paste("The files parameter must be a list of strings."))
+
+    ## normalize the folder path
+    folderPath <- encodeFolderPath(folderPath)
+
+    params <- list(taskId = taskId, protocolName = protocolName, path = path, file = files)
+
+    url <- paste(baseUrl, "pipeline-analysis", folderPath, "getFileStatus.api", sep="")
+    response <- labkey.post(url, toJSON(params, auto_unbox=TRUE))
+
+    return (fromJSON(response, simplifyVector=FALSE, simplifyDataFrame=FALSE))
+}
+
+labkey.pipeline.startAnalysis <- function(baseUrl=NULL, folderPath, taskId, protocolName, path, files,
+    fileIds = list(), pipelineDescription = NULL, protocolDescription = NULL,
+    jsonParameters = NULL, xmlParameters = NULL, allowNonExistentFiles = FALSE, saveProtocol = TRUE)
+{
+    baseUrl=labkey.getBaseUrl(baseUrl)
+
+    ## check required parameters
+    if (missing(baseUrl) || is.null(baseUrl)) stop (paste("A value must be specified for baseUrl."))
+    if (missing(folderPath)) stop (paste("A value must be specified for folderPath."))
+    if (missing(taskId) || is.null(taskId)) stop (paste("A value must be specified for taskId."))
+    if (missing(protocolName) || is.null(protocolName)) stop (paste("A value must be specified for protocolName."))
+    if (missing(path) || is.null(path)) stop (paste("A value must be specified for path."))
+    if (missing(files) || is.null(files)) stop (paste("A value must be specified for files."))
+
+    ## check parameter types
+    if (!is.list(files)) stop (paste("The files parameter must be a list of strings."))
+    if (!is.list(fileIds)) stop (paste("The fileIds parameter must be a list of data ids."))
+    if (!is.null(xmlParameters) && !is.character(xmlParameters))
+        stop (paste("The xml configuration is deprecated, please use the jsonParameters option to specify your protocol description."))
```

Comment:
wouldn't this stop execution of the script? is there a way to log a warning but continue?

Reply:
I was just matching the JS code here where that code throws an error for this case. But I see what you are saying that we wouldn't want to stop the script execution for this deprecated case. Since this is a new API, at least this case won't exist in any existing code so we aren't breaking any backwards compatibility.